File: app/src/main/java/uk/co/ourfriendirony/medianotifier/clients/ArtistClient.java
package uk.co.ourfriendirony.medianotifier.clients;

import java.io.IOException;
import java.util.List;

import uk.co.ourfriendirony.medianotifier.clients.musicbrainz.MusicBrainzClient;
import uk.co.ourfriendirony.medianotifier.mediaitem.MediaItem;

public class ArtistClient implements Client {

    @Override
    public List<MediaItem> searchMediaItem(String name) throws IOException {
        return new MusicBrainzClient().queryArtist(name);
    }

    @Override
    public MediaItem getMediaItem(String id) throws IOException {
        return new MusicBrainzClient().getArtist(id);
    }
}

File: app/src/main/kotlin/uk/co/ourfriendirony/medianotifier/clients/ArtistClient.kt
package uk.co.ourfriendirony.medianotifier.clients

import uk.co.ourfriendirony.medianotifier.clients.musicbrainz.MusicBrainzClient
import uk.co.ourfriendirony.medianotifier.mediaitem.MediaItem
import java.io.IOException

class ArtistClient : Client {
    @Throws(IOException::class)
    override fun searchMediaItem(name: String?): List<MediaItem?> {
        return MusicBrainzClient().queryArtist(name)
    }

    @Throws(IOException::class)
    override fun getMediaItem(id: String?): MediaItem {
        return MusicBrainzClient().getArtist(id)
    }
}